### PR TITLE
HW03 by Ilia Aksenov

### PR DESCRIPTION
**Pull request by benjackson391**

### CODE REVIEW

File: HW03/Ilia_Aksenov/ilia_aksenov_hw_03_t_01.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+require 'time'
+
+class Person
+  attr_accessor :nane, :surname
```
Comment (violettaMi) [inline HW03/Ilia_Aksenov/ilia_aksenov_hw_03_t_01.rb:4]: 😅 
```suggestion
  attr_accessor :name, :surname
```

File: HW03/Ilia_Aksenov/ilia_aksenov_hw_03_t_01.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  def initialize(name:, surname:)
+    @name = name
+    @surname = surname
+  end
+end
+
+class Student < Person
```
Comment (violettaMi) [inline HW03/Ilia_Aksenov/ilia_aksenov_hw_03_t_01.rb:12]: can you split the file into separate files for each class?

File: HW03/Ilia_Aksenov/ilia_aksenov_hw_03_t_01.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    super
+    @homeworks = {}
+  end
+
+  def submit_homework!(homework = '')
+    @homeworks[Time.now.to_i] = homework
+  end
+end
+
+class Mentor < Person
```
Comment (B0RlS) [inline HW03/Ilia_Aksenov/ilia_aksenov_hw_03_t_01.rb:25]: Delete this file if you do not need it. Under 1 class - 1 file.

File: HW03/Ilia_Aksenov/mentor.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+require 'time'
+
+class Mentor
```
Comment (B0RlS) [inline HW03/Ilia_Aksenov/mentor.rb:3]: Why did you implement the Person class if you do not use it?

File: HW03/Ilia_Aksenov/mentor.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    @subscribe_dates[student.object_id] = Time.new.to_i
+  end
+
+  def notifications
+    homeworks = []
+    @students.each do |student|
+      time = @subscribe_dates[student.object_id]
+      homeworks.concat student.homeworks.select { |k| k > time }.values
+    end
+    p homeworks
```
Comment (B0RlS) [inline HW03/Ilia_Aksenov/mentor.rb:23]: Better return homework instead of printing it

File: HW03/Ilia_Aksenov/student.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+require 'time'
+
+class Student
```
Comment (B0RlS) [inline HW03/Ilia_Aksenov/student.rb:3]: same problem

**Comment by B0RlS** [comment]
Mess in commits

File: HW03/Ilia_Aksenov/mentor.rb (hunk 4 above)
Comment (benjackson391) [reply HW03/Ilia_Aksenov/mentor.rb:3]: I tried optimize code, but  I didn't get any profit from it

**Pull request merged**